### ISSUE TITLE
[Question] Why does it only work with plain object?

### ISSUE THREAD

**Issue by tylkomat** [mesqueeb/filter-anything#6]
Why is there a check for plain object? Any class with public properties should work as well.

**Comment by mesqueeb** [comment]
I'm afraid of breaking custom class instances. Because there are so many possibilities. But maybe that's not something I should worry about. 

Do you have a specific example for me ?

**Comment by tylkomat** [comment]
Since it's made for TypeScript and not just JS I was expecting that it would work with classes as well.

I have a class with lots of properties:
```typescript
export class Wood implements WoodInterface {
  "dateRequested": Date;
  "dateAccepted": Date;
  "state": string;
  "scientificName": string;
  "commonName": string;
  "family": string;
  "genus": string;
  "origin": string;
  "difficulties": string;
  "woodType": string;
  "maxHeight": number;
  "maxDiameter": number;
  "age": number;
  "rootsType": string;
  "crown": Array<any>;
  "usage": any;
  "hasStolon": boolean;
  "coppiceAbility": string;
  "growthRate": string;
  "occurence": string;
  "livingArea": Array<any>;
  "landTreesAffiliation": Array<any>;
  "waterTreesAffiliation": Array<any>;
  "ecologicalSignificance": string;
  "siteSuitability": any;
  "climaticRequirements": ClimaticRequirements;
  "ellenbergValue": EllenbergValue;
  "landscapeAesthetic": LandscapeAesthetic;
  "culturalSignificance": CulturalSignificance;
  "images": Array<any>;
  "id": any;
  constructor(data?: WoodInterface) {
    Object.assign(this, data);
  }
}
```
I wanted only some specific properties: `pick(wood, ['id', 'commonName', 'scientificName', 'usage.avgDensity'])` and I received the same object again.

(Which most the most confusing part for me, an Exception might have been better. Now I first had to dig into the source code)

Since it's only 4 properties I did it manually in the end, but a standardized way would be nice.

**Comment by mesqueeb** [comment]
@tylkomat
the reason I want to avoid it working with class instances, is because often class instances have special methods in places to modify its inner properties.
Eg. you wouldn't want to pick or omit props of a `Date` instance directly, it would break the instance.

In your case, I believe it's better to create a method called `getPickedProps` and use my `pick` function inside that method.  It's easy to also share such a method over many interfaces. This is the best way to do it I believe!

**Comment by mesqueeb** [comment]
TLDR;
 I'm tempted to not allow instances, unless you can really convince me : P

**Comment by tylkomat** [comment]
@mesqueeb The problem in my case is that it this code generated from swagger. Updating the generator, which is not mine, would be an option, but just using your funktion would be much simpler.

I think it would not do any harm in your case to put a disclaimer that POJOs (whatever the name is in Typescript universe) are supported, but there are limitations so use at your own risk.

**Comment by mesqueeb** [comment]
@tylkomat try this 

```js
pick({...wood}, ['id', 'commonName', 'scientificName', 'usage.avgDensity'])
```

**Comment by mesqueeb** [comment]
@tylkomat did this solve your issue?
let me know if there's anything else!
